server/routes/v1/upload.ts

```diff
@@ -102,12 +103,16 @@ export const postUpload = [
 
       let version
       try {
-        version = await createVersion(req.user!, {
-          version: metadata.highLevelDetails.modelCardVersion,
-          metadata: metadata,
-        })
+        version = await createVersion(
+          req.user!,
+          {
+            version: metadata.highLevelDetails.modelCardVersion,
+            metadata: metadata,
+          },
+          modelUuid
+        )
       } catch (err: any) {
-        if (err.toString().includes('duplicate key error')) {
+        if (err.code === 400) {
```

Comment:
Sorry, clearly didn't mean code.  I was vaguely expecting the code `E11000` we use elsewhere (and is documented https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.yml).  Which is the unique error code so we only catch the duplicate error here.  It should be somewhere on the response error we get?

Also, if it isn't this error we want to rethrow the error, e.g.:

```
} catch(err: any) {
  if (err.some.code === 'E11000') {
    return res.status(409.json({ ... })
  }

  throw err
}
```